addon/-private/system/store.js

```diff
@@ -800,63 +789,67 @@ Store = Service.extend({
     }
 
     this._pendingFetch.forEach(this._flushPendingFetchForType, this);
-    this._pendingFetch = Map.create();
+    this._pendingFetch.clear();
   },
 
-  _flushPendingFetchForType(pendingFetchItems, typeClass) {
-    var store = this;
-    var adapter = store.adapterFor(typeClass.modelName);
-    var shouldCoalesce = !!adapter.findMany && adapter.coalesceFindRequests;
-    var records = Ember.A(pendingFetchItems).mapBy('record');
+  _flushPendingFetchForType(pendingFetchItems, modelClass) {
+    let store = this;
+    let adapter = store.adapterFor(modelClass.modelName);
+    let shouldCoalesce = !!adapter.findMany && adapter.coalesceFindRequests;
+    let totalItems = pendingFetchItems.length;
+    let internalModels = new Array(totalItems);
+    let seeking = new EmptyObject();
+
+    for (let i = 0; i < totalItems; i++) {
+      internalModels[i] = pendingFetchItems[i].internalModel;
+      seeking[internalModels[i].id] = pendingFetchItems[i];
```

Comment:
I debated on this, the alternative is:

```
let pendingItem = pendingFetchItems[i];
let internalModel = pendingItem.internalModel;
internalModels[i] = internalModel;
seeking[internalModel.id] = pendingItem;
```

This may be more clear but I'm not sure the "cache" actually is any simpler.